### PR TITLE
[Perf] Fuse moss sampling kernel

### PR DESCRIPTION
**Pull request by ppppqp**
## Motivation

Reduce per-token sampling overhead in MOSS-TTS by replacing the two-candidate text sampling operator chain with a fused Triton kernel. Kernel is deliberatedly written to match with eager pytorch implementation. Improved performance with reduced memory movement & kernel launch overhead. 

## Modifications

- Fuse temperature, top-k/top-p filtering, greedy fallback, seeded Gumbel sampling, and token-ID mapping.
- Preserve the existing eager implementation as a fallback.
- Extract some triton sampling utils to shared dir

## Related Issues

N/A

## Accuracy Test

The fused kernel matches the eager implementation exactly across a 4,096-row randomized GPU parity test covering varied logits, temperatures, top-k/top-p settings, seeds, and positions.
One call out is that only top-1 is tested for randomized fuzzy test, because top-k has the order-stability issue in Pytorch that we can not reliably test (pytest complains the discrepancy due to the non-determinstic ordering). However, verified offline that top-k is also in parity with eager version.

## Benchmark & Profiling

**Microbench over the kernel**
Tested on GPU: NVIDIA RTX PRO 6000. Microbench code not included in this PR.
20x speedup kernel wise.
```
batch      eager_us      fused_us   speedup  eager_p95_us  fused_p95_us
    1       231.189        12.528     18.45x       233.310        12.539
    2       235.968        12.498     18.88x       236.645        12.631
    4       236.897        12.459     19.01x       237.815        12.472
    8       237.878        12.486     19.05x       257.644        12.533
   16       256.768        12.434     20.65x       260.145        12.653
   32       253.936        12.512     20.30x       258.134        12.547
   64       239.107        12.380     19.31x       247.216        12.548
```

**e2e benchmark**
Benched with command below on RTX PRO 6000. +10% performance improvement across the board.
```
  python -m sglang_omni.cli serve \
    --model-path OpenMOSS-Team/MOSS-TTS-v1.5 \
    --config examples/configs/moss_tts.yaml \
    --port 8000 \
    --allowed-local-media-path /tmp


  python -m benchmarks.eval.benchmark_tts_seedtts \
    --generate-only \
    --use-existing-server \
    --meta zhaochenyang20/seed-tts-eval-arrow \
    --model OpenMOSS-Team/MOSS-TTS-v1.5 \
    --port 8000 \
    --ref-format references \
    --token-count auto \
    --max-concurrency 8 \
    --warmup 4 \
    --max-samples 50 \
    --lang en \
    --output-dir results/moss_fused_c8_run2
```

Metric | Fused kernel | Eager | Difference (B2 − B1) | Change
-- | -- | -- | -- | --
Model | OpenMOSS-Team/MOSS-TTS-v1.5 | OpenMOSS-Team/MOSS-TTS-v1.5 | — | —
Concurrency | 8 | 8 | 0 | 0.00%
Completed requests | 50 | 50 | 0 | 0.00%
Failed requests | 0 | 0 | 0 | —
Latency mean (s) | 3.907 | 4.430 | +0.523 | +13.39%
Latency median (s) | 3.807 | 4.425 | +0.618 | +16.23%
Latency p95 (s) | 4.997 | 5.606 | +0.609 | +12.19%
Latency p99 (s) | 5.291 | 5.943 | +0.652 | +12.32%
RTF mean | 0.9783 | 1.1051 | +0.1268 | +12.96%
RTF median | 0.9401 | 1.0872 | +0.1471 | +15.65%
RTF p95 | 1.5251 | 1.6676 | +0.1425 | +9.34%
RTF p99 | 1.7172 | 1.8949 | +0.1777 | +10.35%
Audio duration mean (s) | 4.301 | 4.301 | 0.000 | 0.00%
Audio throughput (s/s) | 8.589 | 7.669 | −0.920 | −10.71%
Output throughput (tok/s) | 173.3 | 154.7 | −18.6 | −10.73%
Output tokens/request-s | 58.7 | 44.2 | −14.5 | −24.70%
Output tokens (mean) | 87 | 87 | 0 | 0.00%
Output tokens (total) | 4338 | 4338 | 0 | 0.00%
Prompt tokens (mean) | 166 | 166 | 0 | 0.00%
Prompt tokens (total) | 8286 | 8286 | 0 | 0.00%
Throughput (req/s) | 1.997 | 1.783 | −0.214 | −10.72%

## Checklist

- [x] Format your code according with pre-commit.
- [x] Add unit tests.
- [ ] Update documentation / docstrings / example tutorials as needed.
- [x] Provide throughput / latency benchmark results and accuracy evaluation results as needed.
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.

### CODE REVIEW

File: sglang_omni/models/moss_tts/sampling_kernels.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        active_top_p = (top_p > 0.0) & (top_p < 1.0)
+        max_score = tl.maximum(score0, score1)
+        exp0 = tl.exp(score0 - max_score)
+        exp1 = tl.exp(score1 - max_score)
+        leading_prob = tl.maximum(exp0, exp1) / (exp0 + exp1)
+        remove_lower = active_top_p & (leading_prob > top_p)
+        remove0_top_p = remove_lower & (score0 < score1)
+        remove1_top_p = remove_lower & (score1 <= score0)
+        score0 = tl.where(remove0_top_p, -float("inf"), score0)
+        score1 = tl.where(remove1_top_p, -float("inf"), score1)
```
[the remark this answers is not in the document]
Comment (ppppqp) [reply sglang_omni/models/moss_tts/sampling_kernels.py:70]: No need to fix imo because I think stable sorting is disabled for performance, and the tie-breaking is trivial behavioral consideration.
